src/analysis/retail/warlock/demonology/modules/talents/SoulStrike.tsx

```diff
@@ -25,24 +28,55 @@ class SoulStrike extends Analyzer {
       Events.damage.by(SELECTED_PLAYER_PET).spell(SPELLS.SOUL_STRIKE_DAMAGE),
       this.handleSoulStrikeDamage,
     );
+    this.addEventListener(
+      Events.resourcechange.by(SELECTED_PLAYER_PET).spell(SPELLS.SOUL_STRIKE_SHARD_GEN),
+      this.handleSoulStrikeShardGen,
+    );
   }
 
   handleSoulStrikeDamage(event: DamageEvent) {
     this.damage += event.amount + (event.absorbed || 0);
   }
 
+  handleSoulStrikeShardGen(event: ResourceChangeEvent) {
+    if (event.resourceChangeType === RESOURCE_TYPES.SOUL_SHARDS.id) {
+      this.totalCasts += 1;
+      // Check if player was at shard cap (5 shards = 50 fragments)
+      // The event shows the state AFTER the resource change, so we need to check if waste > 0
+      if (event.waste > 0) {
+        this.wastedShards += 1;
+      }
+    }
+  }
+
+  get shardsGenerated() {
+    return this.totalCasts;
```

Comment:
Do you want this to perhaps show "good" shards generated rather than all shards when you're showing the wasted shards separately?

Reply:
good catch, thanks